emma-language/src/main/scala/org/emmalanguage/ast/AST.scala

```diff
@@ -90,6 +90,30 @@ trait AST extends CommonAST
     case api.TypeQuote(tpe) => api.Type.tree(tpe)
   }.andThen(_.tree)
 
+  /**
+   * Rewrites `A.this.x` as `x` - can be used to overcome.
+   *
+   * Can be used before `showCode` to avoid printing invalid trees such as
+   *
+   * {{{
+   * class A {
+   *   val x = 42;
+   *   class A {
+   *     scala.Predef.println(A.this.x)
+   *   }
+   * }
+   * }}}
+   *
+   * '''Use with caution''': the transformation is not sound and can break due to variable
+   * capture, e.g. if `x` is defined in the inner `A` in the case above.
```

Comment:
Capture happens when `x` is a local variable, e.g.
```scala
class A {
  val x = 42
  def method: {
    val x = "42"
    // prints 421 instead of 43
    println(this.x + 1)
  }
}